sklearn/model_selection/tests/test_split.py

```diff
@@ -1026,6 +1026,27 @@ def test_train_test_split_errors():
     assert_raises(ValueError, train_test_split, range(3), range(42))
     assert_raises(ValueError, train_test_split, range(10),
                   shuffle=False, stratify=True)
+    assert_raises(ValueError, train_test_split, range(10), train_size=1.2)
```

Comment:
This case will raise `FutureWarning` while running test case against bleeding edge release/dependencies with warnings enabled

https://github.com/scikit-learn/scikit-learn/blob/4ca4bce1082e33dc7bb0859dd703afa4aeb6a4ab/sklearn/model_selection/_split.py#L2179-L2185

Travis CI error: https://travis-ci.org/scikit-learn/scikit-learn/jobs/465113920#L2872

Reply:
from version 0.21 `train_test_split` takes either test_size or both, probably you could use `ignore_warnings` to pass the failing test case like 

https://github.com/scikit-learn/scikit-learn/blob/654c452291996c019158ab00afd07a712a9d2901/sklearn/tests/test_common.py#L74-L79

Reply:
Thanks for the tip @nixphix. Is this what you have in mind as the fix:

```python
@ignore_warnings
def test_train_test_split_errors():
```

Reply:
not exactly that would ignore all warnings, I would suggest using context manager like
```
with ignore_warnings(category=(FutureWarning)):
    ...
```

Reply:
you have an example in this very test:
```python
with warnings.catch_warnings():
         # JvR: Currently, a future warning is raised if test_size is not
         # given. As that is the point of this test, ignore the future warning
         warnings.filterwarnings("ignore", category=FutureWarning)
         assert_raises(ValueError, train_test_split, range(3), train_size=1.1)
```
But maybe a better way would just be to set the test_size parameter.

Reply:
I pushed an update. I hope this is the last fix